src/payment/models/pricing/utils/price-path.ts

```diff
@@ -0,0 +1,76 @@
+import { cloneDeep } from 'lodash';
+import { Price } from '../../exchange/dto/price.dto';
+import { PriceRequest, PriceResult, PriceStepResult } from '../interfaces';
+import { PricingPathAlias } from '../services/pricing.service';
+import { PricePathInitSpecification } from '../specifications/price-path-init.specification';
+import { PriceStep } from './price-step';
+
+export class PricePath {
+  constructor(public readonly alias: PricingPathAlias, private readonly steps: PriceStep[]) {
+    PricePathInitSpecification.isSatisfiedBy(this);
+  }
+
+  async execute(request: PriceRequest): Promise<PriceResult> {
+    const results: PriceStepResult[] = [];
+
+    for (let i = 0; i < this.steps.length; i++) {
+      const step = this.steps[i];
+
+      if (this.steps.length === 1) {
+        results.push(await step.execute({ from: request.from, to: request.to }));
+        break;
+      }
+
+      if (i === 0) {
+        results.push(await step.execute({ from: request.from }));
+        continue;
+      }
+
+      if (i === this.steps.length - 1) {
+        results.push(await step.execute({ to: request.to }));
+        continue;
+      }
+
+      results.push(await step.execute());
```

Comment:
Would perhaps be easier with if-else instead of break/continue.

Reply:
would still need to `continue` so that last line is not triggered, and with `else if` it reads not as clean imho. Would keep it like that tbh